scene/main/viewport.cpp

```diff
@@ -436,6 +453,18 @@ int Viewport::_sub_window_find(Window *p_window) {
 void Viewport::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_ENTER_TREE: {
+#ifdef TOOLS_ENABLED
```

Comment:
ViewportTexture is `resource_local_to_scene` enabled by default and is required to be in the same scene as the target Viewport. At runtime, the scene root is difficult to determine. So it's hard to update this `viewport_path` relative to the scene root.

Reply:
Can `ViewportTexture` disable `resource_local_to_scene`? ViewportTexture won't work when `resource_local_to_scene` is disabled. The disabled case never seems to be considered.

Reply:
I remember there was an error/warning when it was disabled.

Reply:
At runtime, it may be more convenient to expose a method for getting the target Viewport. Because it is impossible to determine how to move nodes and how to pack a new scene.

Reply:
This sounds like more involved changes, while users don't seem to have such problems. I only suggested changing the ifdef, because it could be done basically for free. Anything more is not necessary for now.